---
title: "MUC5AC Fine-mapping analysis"
author: "Nomei Nicole Piga"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    code_folding: "hide"
    toc: true
    toc_float: true
    theme: united
---


```{r, setup, hide = TRUE}
#library reticulate to be able to add chuncks in another language
library(reticulate)
# To run this .rmd file in a terminal being in the project folder:
#export PATH=${PATH}:/cm/shared/apps/R/deps/rstudio/bin/pandoc
#file="./report/Report_muc5a_finemapping.Rmd"
#module unload R/4.2.1
#module load R/4.1.0
#Rscript -e 'rmarkdown::render("'$file'")'
```
# Rationale
Using PolyFun+SuSie to fine-map the MUC5AC region in Shrine et al. moderate-severe asthma and GBMI asthma summary statistics; compare them to each other and the chronic sputum.
<br>
Polyfun is a software that allows the use of prior probabilities as per-SNP heritability calculated based on multiple functional annotations/measures. 
<br>
The posterior probability is calculated using the SNP's marginal effect size (from the GWAS analysis) together with the prior (https://github.com/omerwe/polyfun/issues/131).
##Data
Bayesian fine-mapping was conducted as part of the chronic sputum paper (supplementary table 10).
<br>
The raw GWAS results for the chronic sputum study: /data/gen1/Phlegm/Results_files/GWAS/combined_results_INFO_MAC20
<br>
Shrine et al. moderate-severe asthma summary statistics with sentinel rs11603634: /rfs/TobinGroup/kaf19/Public_data/ukb_mod_sev_asthma/Shrine_30552067_moderate-severe_asthma.txt.gz
<br>
GBMI asthma summary statistics with sentinel variant rs11245964: /rfs/TobinGroup/kaf19/Public_data/gbmi_asthma/Asthma_Bothsex_eur_inv_var_meta_GBMI_052021_nbbkgt1.txt.gz

# Analysis
Code: /home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/polyfun_susie.sh
```{r, code = readLines('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/polyfun_susie.sh'), eval=FALSE}
```
Code: /home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/create_maf.R
```{r, code = readLines('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/create_maf.R'), eval=FALSE}
```
Code: /home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/credset.R
```{r, code = readLines('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/credset.R'), eval=FALSE}
```
Code: /home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/create_maf_gbmi.R
```{r, code = readLines('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/create_maf_gbmi.R'), eval=FALSE}
```
Code: /home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/gbmi_b37.R
```{r, code = readLines('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/gbmi_b37.R'), eval=FALSE}
```
Code: /home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/compare_credset.R
```{r, code = readLines('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/compare_credset.R'), eval=FALSE}
```
Code: /home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/test_polyfun_susie.sh
```{r, code = readLines('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/test_polyfun_susie.sh'), eval=FALSE}
```
Code: /home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/compare_different_params.R
```{r, code = readLines('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/src/compare_different_params.R'), eval=FALSE}
```

Fine-mapping of the MUC5AC regions +/- 500Kb from the sentinel variant: rs11603634 for European moderate-to-severe, rs11245964 for European GBMI all-comer asthma.
<br>
For GBMI, the sample size differs for SNP. I am still trying to figuring it out how to use polyfun+SuSie with a per-SNP sample size. I found an R package that allows this, but I am unable to download it in R due to memory problem with github. I used the median sample size (which is also the maximum number) and the minimum sample size for now. In both cases, the software found the same variants in the 95% credible set, and the posterior inclusion probabilities are similar.

# Results
A total of 3690 variants were examined for the moderate-to-severe fine-mapping and 34 were identified in 95% credible set.
<br>
Regarding GBMI, 3,781 variants were investigated and 11 variants were identified for the 95% credible set.
<br>
```{r, echo = FALSE}
df <- read.table('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/output/credset_3_gwas.txt')
DT::datatable(df, height = 200, options = list(scrollX = TRUE, scrollY = FALSE))
```
<br>
<br>
<br>
Comparing the credible set, 6 variants were shared across all the three credible sets; 21 variants were shared among the chronic sputum and the moderate-to-severe; 3, 7, and 5 variants were private respectively to chronic sputum, moderate-to-severe, and GBMI (Figure 1).
<br>
```{r, out.width='80%', out.height='80%', fig.align='center', fig.cap='Figure 1. Venn diagram of the three 95% credible sets.', echo = FALSE}
knitr::include_graphics('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/output/Venn_asthma_credset.png')
```

# Notes
I used the pre-computed prior SNP probabilities calculated by polyfun's authors. In doing so, I lost 441 and 389 variants in the fine-mapping for moderate-to-severe and GBMI study respectively, among which two variants that were found in the chronic sputum credible set (10.67% and 9.33% missing variants).
<br>
To notice as well, polyfun's author pointed out that results from meta-analysis may not be reliable (Table 3 in the main paper [1]) (https://github.com/omerwe/polyfun/issues/161).
<br>
In addition, I tried to use different parameters for minimum info score and maximum number of causal variants: info score 0.90, and 1 causal variant.
<br>
For moderate-to-severe, all the variants are shared except for two (Figure 2):
```{r, out.width='100%', out.height='100%', fig.align='center', fig.cap='Figure 2. Venn diagram of the credible sets for different parameters for moderate-to-severe.', echo = FALSE}
knitr::include_graphics('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/output/Venn_asthma_credset_modsev_params.png')
```
<br>
For GBMI all-comer asthma, having a different type of maximum causal variants influenced which variants are in the credible set (Figure 3 and 4):
```{r, out.width='100%', out.height='100%', fig.align='center', fig.cap='Figure 3. Venn diagram of the credible sets for different parameters for GBMI, with N = median sample size.', echo = FALSE}
knitr::include_graphics('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/output/Venn_asthma_credset_gbmi_Nmedian_params.png')
```
<br>
```{r, out.width='100%', out.height='100%', fig.align='center', fig.cap='Figure 4. Venn diagram of the credible sets for different parameters for GBMI, with N = minimum sample size.', echo = FALSE}
knitr::include_graphics('/home/n/nnp5/PhD/PhD_project/muc5a_finemapping/output/Venn_asthma_credset_gbmi_Nmin_params.png')
```

# Conclusion
A different minimum threshold for info score did not influence the causal variants in a consistent way. Althouugh, a different numbe of maximum causal variant showed visible differences in terms of variants in the credible set for GBMI analysis. As well, fine-mapping from meta-analysis may result in less reliable data, as per polyfun's author.
<br>
When using info score 0.85, and 10 possible causal variants, a total of 42 variants were identified in the credible sets among the three studies where ~64% showed were found by at least two studies.  


# References
1.Weissbrod 0. et al, Functionally informed fine-mapping and polygenic localization of complex trait heritability. Nat Genet 2020,doi:https://doi.org/10.1038/s41588-020-00735-5
<br>
https://github.com/omerwe/polyfun



